src/Files.Uwp/Helpers/LayoutPreferences/LayoutPreferences.cs

```diff
@@ -0,0 +1,72 @@
+        public SortDirection DirectorySortDirection;
+        public bool SortDirectoriesAlongsideFiles;
+        public GroupOption DirectoryGroupOption;
+        public FolderLayoutModes LayoutMode;
+        public int GridViewSize;
+        public bool IsAdaptiveLayoutOverridden;
+
+        public ColumnsViewModel ColumnsViewModel;
+
+        [LiteDB.BsonIgnore]
+        public static LayoutPreferences DefaultLayoutPreferences => new LayoutPreferences();
+
+        public LayoutPreferences()
+        {
+            this.LayoutMode = UserSettingsService.LayoutSettingsService.DefaultLayoutMode;
+            this.GridViewSize = UserSettingsService.LayoutSettingsService.DefaultGridViewSize;
+            this.DirectorySortOption = UserSettingsService.LayoutSettingsService.DefaultDirectorySortOption;
+            this.DirectoryGroupOption = UserSettingsService.LayoutSettingsService.DefaultDirectoryGroupOption;
+            this.DirectorySortDirection = UserSettingsService.LayoutSettingsService.DefaultDirectorySortDirection;
+            this.SortDirectoriesAlongsideFiles = UserSettingsService.LayoutSettingsService.DefaultSortDirectoriesAlongsideFiles;
+            this.IsAdaptiveLayoutOverridden = false;
+
+            this.ColumnsViewModel = new ColumnsViewModel();
+            this.ColumnsViewModel.DateCreatedColumn.UserCollapsed = !UserSettingsService.LayoutSettingsService.ShowDateCreatedColumn;
+            this.ColumnsViewModel.DateModifiedColumn.UserCollapsed = !UserSettingsService.LayoutSettingsService.ShowDateColumn;
+            this.ColumnsViewModel.ItemTypeColumn.UserCollapsed = !UserSettingsService.LayoutSettingsService.ShowTypeColumn;
+            this.ColumnsViewModel.SizeColumn.UserCollapsed = !UserSettingsService.LayoutSettingsService.ShowSizeColumn;
+            this.ColumnsViewModel.TagColumn.UserCollapsed = !UserSettingsService.LayoutSettingsService.ShowFileTagColumn;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+            {
+                return false;
+            }
+            if (obj == this)
+            {
+                return true;
+            }
+            if (obj is LayoutPreferences prefs)
+            {
+                return (
+                    prefs.LayoutMode == this.LayoutMode &&
+                    prefs.GridViewSize == this.GridViewSize &&
+                    prefs.DirectorySortOption == this.DirectorySortOption &&
+                    prefs.DirectorySortDirection == this.DirectorySortDirection &&
+                    prefs.SortDirectoriesAlongsideFiles == this.SortDirectoriesAlongsideFiles &&
+                    prefs.IsAdaptiveLayoutOverridden == this.IsAdaptiveLayoutOverridden &&
+                    prefs.ColumnsViewModel.Equals(this.ColumnsViewModel));
+            }
+            return base.Equals(obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
+        }
```

Comment:
Do we need this for DB stuff? In that case, maybe overriding the hashcode based on its own properties would be better since it would also ensure we adhere to the equals-hashcode contract. Having equals return true while the hashcodes are different might behave strangely.

Reply:
It's used for this check, to avoid storing layout prefs in the DB if they match the default ones:
```cs
if (DbInstance.GetPreferences(folderPath, frn) is null)
{
    if (LayoutPreferences.DefaultLayoutPreferences.Equals(prefs))
    {
        return; // Do not create setting if it's default
    }
}
DbInstance.SetPreferences(folderPath, frn, prefs);
```
Not added GetHasCode() implementation, the way Resharper does it.